src/main.ts

```diff
@@ -313,6 +318,79 @@ class NixInstallerAction {
     }
   }
 
+  async setup_flakehub(): Promise<number> {
+    const jwt = await actions_core.getIDToken("api.flakehub.com");
+
+    const spawned = spawn(
+      `sudo`,
+      ["sh", "-c", "mkdir -p /etc/nix && tee -a /etc/nix/netrc"],
```

Comment:
How about a `--netrc` arg on the installer? I don't really want to shell out for this.

Otherwise, I suggest writing this data using `nodejs`'s standard library instead of using shell commands, then using `sudo` to move the file. Here's an example of writing a binary out from this repo: https://github.com/DeterminateSystems/nix-installer-action/blob/3ebd1aebb47f95493b62de6eec0cac3cd74e50a9/src/main.ts#L363-L401

Alternatively, we can rig up a way for this to call a specific function as root to do this without shelling out.

Reply:
We chatted on Discord and decided it'd be better to write out a separate netrc and pass in a netrc-file option.